pkg/controller/ovsnodeosp/ovsnodeosp_controller.go

```diff
@@ -0,0 +1,323 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch ConfigMaps owned by OVSNodeOsp
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
+		IsController: false,
+		OwnerType:    &neutronv1.OVSNodeOsp{},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner OVSNodeOsp
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &neutronv1.OVSNodeOsp{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileOVSNodeOsp implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileOVSNodeOsp{}
+
+// ReconcileOVSNodeOsp reconciles a OVSNodeOsp object
+type ReconcileOVSNodeOsp struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a OVSNodeOsp object and makes changes based on the state read
+// and what is in the OVSNodeOsp.Spec
+func (r *ReconcileOVSNodeOsp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling OVSNodeOsp")
+
+	// Fetch the OVSNodeOsp instance
+	instance := &neutronv1.OVSNodeOsp{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// ScriptsConfigMap
+	scriptsConfigMap := ovsnodeosp.ScriptsConfigMap(instance, instance.Name+"-scripts")
+	if err := controllerutil.SetControllerReference(instance, scriptsConfigMap, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+	// Check if this ScriptsConfigMap already exists
+	foundScriptsConfigMap := &corev1.ConfigMap{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: scriptsConfigMap.Name, Namespace: scriptsConfigMap.Namespace}, foundScriptsConfigMap)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Creating a new ScriptsConfigMap", "ScriptsConfigMap.Namespace", scriptsConfigMap.Namespace, "Job.Name", scriptsConfigMap.Name)
+		err = r.client.Create(context.TODO(), scriptsConfigMap)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else if !reflect.DeepEqual(scriptsConfigMap.Data, foundScriptsConfigMap.Data) {
+		reqLogger.Info("Updating ScriptsConfigMap")
+		scriptsConfigMap.Data = foundScriptsConfigMap.Data
+	}
+
+	scriptsConfigMapHash, err := util.ObjectHash(scriptsConfigMap.Data)
```

Comment:
is this used?

Reply:
See @stuggi's comment above. If we pass this into the env for the daemonset (or wherever it is applied) it will trigger an update there if a change is detected. Like if we update the scripts or something